File: TP1/Exo 1.2/Python/cercle.py
# Importation de la bibliothèque math pour utiliser des fonctions mathématiques, notamment pi
import math

# Fonction qui calcule l'aire d'un cercle en fonction de son rayon
def calcul_aire(rayon):
    # Récupération de la valeur de pi depuis la bibliothèque math
    pi = math.pi
    
    # Calcul de l'aire : π * rayon²
    aire = pi * rayon * rayon

    # Affichage de l'aire calculée, formatée avec une précision adéquate
    print(f"\nL'aire est de {aire} cm² !")

# Fonction qui calcule le périmètre (ou circonférence) d'un cercle en fonction de son rayon
def calcul_perimetre(rayon):
    # Calcul du périmètre : 2 * π * rayon
    peri = 2 * math.pi * rayon

    # Affichage du périmètre calculé, formaté avec une précision adéquate
    print(f"Le périmètre est de {peri} cm !")

# Fonction principale pour démarrer le programme
def main():
    # Demande à l'utilisateur de saisir une valeur pour le rayon du cercle, en la convertissant en entier
    rayon = int(input("Saisir une valeur de rayon pour le cercle : "))

    # Appel de la fonction pour calculer et afficher l'aire
    calcul_aire(rayon)
    # Appel de la fonction pour calculer et afficher le périmètre
    calcul_perimetre(rayon)

# Point d'entrée du programme
if __name__ == "__main__":
    main()

File: TP1/Exo 1.2/C/cercle.c
#include <stdio.h>
#include <math.h> // Inclusion des bibliothèques standard nécessaires

// Définition de la constante M_PI (valeur de π)
#define M_PI 3.14159265358979323846

// Fonction qui calcule l'aire d'un cercle à partir de son rayon
int calcul_aire(float rayon)
{
    // Calcul de l'aire : π * rayon²
    int aire = M_PI * rayon * rayon;

    // Affichage de l'aire calculée
    printf("La valeur de l'aire est %d\n", aire);

    return 0; // Retourne 0 (signifiant la fin de l'exécution correcte de la fonction)
}

// Fonction qui calcule le périmètre (ou circonférence) d'un cercle à partir de son rayon
int calcul_peri(float rayon)
{
    // Calcul du périmètre : 2 * π * rayon
    int peri = 2 * M_PI * rayon;

    // Affichage du périmètre calculé
    printf("La valeur du périmètre est %d\n", peri);

    return 0; // Retourne 0 (signifiant la fin de l'exécution correcte de la fonction)
}

// Fonction principale (point d'entrée du programme)
int main()
{
    float rayon; // Déclaration de la variable pour stocker le rayon
    printf("Entrez une valeur pour le rayon : ");
    
    // Boucle pour s'assurer que l'utilisateur entre une valeur valide
    while(scanf("%f",&rayon) != 1)
    {
        printf("Entrez une valeur pour le rayon : ");
        // Vide le buffer d'entrée pour éviter une boucle infinie en cas d'entrée invalide
        while(getchar() != '\n');
    }
    
    // Appel de la fonction pour calculer et afficher l'aire
    calcul_aire(rayon);
    // Appel de la fonction pour calculer et afficher le périmètre
    calcul_peri(rayon
